data/events/scripts/player.lua

```diff
@@ -199,6 +199,11 @@ function Player:onLook(thing, position, distance)
 			end
 		end
 	end
+		if LOOK_MARRIAGE_DESCR and thing:isCreature() then
+        if thing:isPlayer() then
+            description = description .. self:getMarriageDescription(thing)
```

Comment:
self:getMarriageDescription() is calling getPlayerSpouse(), getPlayerMarriageStatus() and getPlayerNameById()
3 db queries in a onLook function

I think this function would be way easier to make in player.cpp/h

Reply:
How cool are you going to post or just said?  
kk

Reply:
Sorry, I don't get you. But if you meant if am I going to post some code, no. He asked for a hint/advice to improve the code, and I gave mine.